backend/app/Repositories/EloquentChartSessionsRepository.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\DataTransformer\Sessions\ChartSessions;
+use App\Repositories\Contracts\ChartSessionsRepository;
+use App\Contracts\Common\DatePeriod;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use App\Entities\Session;
+
+final class EloquentChartSessionsRepository implements ChartSessionsRepository
+{
+    private function toTimestamp(string $columnName): string
+    {
+        return "extract(epoch FROM $columnName)";
+    }
+
+    public function findByFilter(DatePeriod $filterData, int $websiteId): Collection
+    {
+        $subQuery = "SELECT s.*" .
+        "FROM sessions AS s " .
+        "WHERE " .
+            "s.website_id = " . "$websiteId AND (" .
+            $this->toTimestamp('s.start_session') . " >= :start_date AND " .
+            $this->toTimestamp('s.start_session') . " <= :end_date) OR (" .
+            $this->toTimestamp('s.start_session') . " <= :start_date AND " .
+            $this->toTimestamp('s.end_session') . " >= :start_date)";
+
+        $result = DB::select((string)$subQuery, [
+        'start_date' => $filterData->getStartDate()->getTimestamp(),
+        'end_date' => $filterData->getEndDate()->getTimestamp(),
+        ]);
+
+        return collect($result)->map(function ($item) {
+            return array(
```

Comment:
removed $item->website_id   ( Leave in case of debug)

ChartValue used in GetSessionsAction, output in the array to be able to sort through the array using foreach

Required to use DTO ChartSessionValue?